pkg/apis/options/providers.go

```diff
@@ -0,0 +1,104 @@
+package options
+
+// Providers is a collection of definitions for providers.
+type Providers []Provider
+
+// Provider holds all provider configuration
+type Provider struct {
+	ClientID         string `json:"clientID,omitempty"`
+	ClientSecret     string `json:"clientSecret,omitempty"`
+	ClientSecretFile string `json:"clientSecretFile,omitempty"`
+
+	KeycloakConfig  KeycloakOptions  `json:"keycloakConfig,omitempty"`
+	AzureConfig     AzureOptions     `json:"azureConfig,omitempty"`
+	BitbucketConfig BitbucketOptions `json:"bitbucketConfig,omitempty"`
+	GitHubConfig    GitHubOptions    `json:"githubConfig,omitempty"`
+	GitLabConfig    GitLabOptions    `json:"gitlabConfig,omitempty"`
+	GoogleConfig    GoogleOptions    `json:"googleConfig,omitempty"`
+	OIDCConfig      OIDCOptions      `json:"oidcConfig,omitempty"`
+	LoginGovConfig  LoginGovOptions  `json:"loginGovConfig,omitempty"`
+
+	ProviderID      string   `json:"providerID,omitempty"`
+	ProviderType    string   `json:"provider,omitempty"`
+	ProviderName    string   `json:"providerDisplayName,omitempty"`
+	ProviderCAFiles []string `json:"providerCAFiles,omitempty"`
+
+	LoginURL          string   `json:"loginURL,omitempty"`
+	RedeemURL         string   `json:"redeemURL,omitempty"`
+	ProfileURL        string   `json:"profileURL,omitempty"`
+	ProtectedResource string   `json:"resource,omitempty"`
+	ValidateURL       string   `json:"validateURL,omitempty"`
+	Scope             string   `json:"scope,omitempty"`
+	Prompt            string   `json:"prompt,omitempty"`
+	ApprovalPrompt    string   `json:"approvalPrompt,omitempty"`
+	AllowedGroups     []string `json:"allowedGroups,omitempty"`
+
+	AcrValues string `json:"acrValues,omitempty"`
+}
+
+type KeycloakOptions struct {
+	KeycloakGroup string `json:"keycloakGroup,omitempty"`
+}
```

Comment:
Just thinking about some refactoring that @NickMeves is doing at the moment to remove a lot of provider specific group options and make these kinds of options fall back to just have a group, I suspect once that is merged we could just do the conversion/fallback in the legacyprovider convert and remove all of the `Group` options from the provider, ergo, we may not need a `KeycloakOptions`. @NickMeves WDYT?

Reply:
I think this was resolved in the rebase, please correct me if I'm wrong